scm/etc/scripts/UFS_case_gen.py

```diff
@@ -2684,62 +3584,78 @@ def main():
     
     #read in arguments
     (location, indices, date, in_dir, grid_dir, forcing_dir, tile, area, case_name, 
-     old_chgres, lam, save_comp, use_nearest) = parse_arguments()
+     old_chgres, lam, save_comp, use_nearest, forcing_method, vertical_method, geos_wind_forcing, wind_nudge) = parse_arguments()
+    
+    #find indices corresponding to both UFS history files and initial condition (IC) files
+    (hist_i, hist_j, hist_lon, hist_lat, hist_dist_min, angle_to_hist_point, neighbors, dx, dy) = find_loc_indices_UFS_history(location, forcing_dir)
+    
+    (IC_i, IC_j, tile, IC_lon, IC_lat, IC_dist_min, angle_to_IC_point) = find_loc_indices_UFS_IC(location, grid_dir, lam, tile, indices)
         
-    #find tile containing the point using the supergrid if no tile is specified 
-    #if not tile and not lam:
-    if not tile:
-        tile = int(find_tile(location, grid_dir, lam))
-        if tile < 0:
-            message = 'No tile was found for location {0}'.format(location)
-            logging.critical(message)
-            raise Exception(message)
-        print('Tile found: {0}'.format(tile))
+    #compare the locations of the found history file point and UFS IC point
+    compare_hist_and_IC_points(location, hist_lon, hist_lat, IC_lon, IC_lat, hist_dist_min, angle_to_hist_point, IC_dist_min, angle_to_IC_point)
     
-    #find index of closest point in the tile if indices are not specified
-    if not indices:
-        (tile_j, tile_i, point_lon, point_lat, dist_min) = find_loc_indices(location, grid_dir, tile, lam)
-        print('The closest point in tile {0} has indices [{1},{2}]'.format(tile,tile_i,tile_j))
-        print('This index has a central longitude/latitude of [{0},{1}]'.format(point_lon,point_lat))
-        print('This grid cell is approximately {0} km away from the desired location of {1} {2}'.format(dist_min/1.0E3,location[0],location[1]))
+    #read in surface data for the intial conditions at the IC point
+    surface_data = get_UFS_surface_data(in_dir, tile, IC_i, IC_j, old_chgres, lam)
+    
+    #when using the advective forcing methods, initial conditions are taken from the history files; check that the history file surface type is the same as the UFS IC point from which data is read
+    check_IC_hist_surface_compatibility(forcing_dir, hist_i, hist_j, surface_data, lam, old_chgres, tile)
+    
+    #read in orographic data for the initial conditions at the IC point
+    oro_data = get_UFS_oro_data(in_dir, tile, IC_i, IC_j, lam)
+    
+    #read in the initial condition profiles
+    
+    if (forcing_method == 1 or forcing_method == 4):
```

Comment:
Is this supposed to be 1 or 3? There is no 4 (yet)

Reply:
Yep.

Reply:
Fixed.

Reply:
This was probably causing the weird/bad results for time=0 for fm=3.

Reply:
I had started using 'fm' to control wind nudging and geostrophic but made them separate controls, but forgot to remove it here. Good catch.

Reply:
Yeah, I left some images on the slack channel for before/after my fix, fm=3 looked a lot better